### DISCUSSION TITLE
tabby-agent build error

### DISCUSSION THREAD

**Discussion by 0xd3bu9** [TabbyML/tabby#2282] [General]
**Describe the bug**
I want to build a **tabby-agent** local (don't use `npm i`).
I ran the build successfully and I put it in locally to use on folder `tabby/clients/vscode`
![image](https://github.com/TabbyML/tabby/assets/108015133/b12965d0-2cd5-482c-a1ef-94de0330d2e8)
I then ran `npm run build` for testing, but they only show the error:
![image](https://github.com/TabbyML/tabby/assets/108015133/592dba6f-b8f8-4a79-b921-7610b19c5c69)

What do I need to do to fix this?

**Information about your version**
  "name": "vscode-tabby",
  "version": "1.5.0-dev",

 "tabby-agent": "1.4.2",

![image](https://github.com/TabbyML/tabby/assets/108015133/ae0a7b94-4163-4c6d-a5ea-46dd1262fa07)

Build environment:
- Windows 11
- NodeJS  v18.19.1
-  "tsup": "^7.1.0",
-  "typescript": "^5.3.2"

**Comment by wsxiaoys** [comment]
js dependencies are managed with `pnpm` - please try:
```
pnpm i
pnpm run build
```

**Comment by 0xd3bu9** [comment]
Hi @wsxiaoys.

This time I used the latest version to rebuild vscode.
But this time it was probably a serious error when I finished building it and couldn't use it.
I did the following steps:
## 1. `git clone`
## 2. cd `tabby-main/client/tabby-agent`, run command `pnpm run build` and get a warning
![image](https://github.com/TabbyML/tabby/assets/108015133/474857ca-e9f2-49ab-99c9-53c01b0882d6)

## 3. Next go to `cd /tabby-chat-pannel` and run the command `pnpm run build`
![image](https://github.com/TabbyML/tabby/assets/108015133/8ad7d00f-9d5f-47c6-8254-be8adfb90291)

## 4. Finally I went back to the `vscode` folder and ran the command `pnpm run build`
![image](https://github.com/TabbyML/tabby/assets/108015133/972d4698-4050-420e-b963-2b1055e9b731)

## 5. When I run it, there is actually an error with nothing displayed
![image](https://github.com/TabbyML/tabby/assets/108015133/a0236676-d081-4a5f-bd9e-52b5039ece3a)



What step did I do wrong, please guide me.
